packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -4118,13 +4121,19 @@ export class ContainerRuntime
 		}
 	}
 
-	private reSubmitBatch(batch: PendingMessageResubmitData[]) {
+	private reSubmitBatch(batch: PendingMessageResubmitData[], batchId: BatchId) {
 		this.orderSequentially(() => {
 			for (const message of batch) {
 				this.reSubmit(message);
 			}
 		});
-		this.flush();
+
+		// Only include Batch ID if "Offline Load" feature is enabled
+		// It's only needed to identify batches across container forks arising from misuse of offline load.
+		const includeBatchId =
+			this.mc.config.getBoolean("Fluid.Container.enableOfflineLoad") ?? false;
```

Comment:
It's interesting that we're reading this flag from two different places. I don't know if that's a good or bad thing. And it always makes me worry about loader-runtime back compat with older drivers and how they might behave.

Reply:
Two different places in this PR?  I thought it was just here.

This flag is read in multiple places and is the single gate for all Offline related scenarios.  Good question about loader/runtime... Yeah... I see what you mean, the other hit is in Loader layer.  I'll follow up, but leaving here for now since none of this has shipped yet.